clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreBufferService.java

```diff
@@ -245,12 +248,23 @@ private static void processRecord(
   private static void processCommand(
       CommandQueueNode cmd,
       StoreIngestionTask ingestionTask,
-      PartitionConsumptionState pcs) {
+      PartitionConsumptionState pcs,
+      PubSubTopicPartition topicPartition) {
     // We only support SYNC_OFFSET command for now.
     if (cmd.getCommandType() != CommandQueueNode.CommandType.SYNC_OFFSET) {
       throw new VeniceException("Unsupported command type: " + cmd.getCommandType());
     }
 
+    if (pcs == null) {
+      cmd.executeSync(() -> {
+        String msg = "PCS for topic-partition: {} is null. Skipping {} command in StoreBufferDrainer.";
+        if (!REDUNDANT_LOGGING_FILTER.isRedundantException(msg)) {
+          LOGGER.info(msg, topicPartition, cmd.getCommandType());
+        }
+      });
```

Comment:
Let's not execute the logging inside `executeSync`. Because in `executeSync`, it will eventually completes an internal Future object indicating that offset sync action is finished. Meanwhile, `REDUNDANT_LOGGING_FILTER` might be an overkill as we are not supposed to see this happening repeatedly.

Reply:
Sorry, I think I am wrong about `executeSync` and your code is correct, but maybe not using  `REDUNDANT_LOGGING_FILTER`?

Reply:
Thank you for your review. I have updated the code, can you help take another look?